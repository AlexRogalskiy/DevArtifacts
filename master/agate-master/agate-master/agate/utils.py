#!/usr/bin/env python
# -*- coding: utf8 -*-

"""
This module contains a collection of utility classes and functions used in
agate.
"""

from collections import OrderedDict, Sequence
from functools import wraps
import string
import warnings
from slugify import Slugify, UniqueSlugify

try:
    from cdecimal import Decimal, ROUND_FLOOR, ROUND_CEILING, getcontext
except ImportError:  # pragma: no cover
    from decimal import Decimal, ROUND_FLOOR, ROUND_CEILING, getcontext

import six


#: Sentinal for use when `None` is an valid argument value
default = object()


def memoize(func):
    """
    Dead-simple memoize decorator for instance methods that take no arguments.

    This is especially useful since so many of our classes are immutable.
    """
    memo = None

    @wraps(func)
    def wrapper(self):
        if memo is not None:
            return memo

        return func(self)

    return wrapper


class NullOrder(object):
    """
    Dummy object used for sorting in place of None.

    Sorts as "greater than everything but other nulls."
    """
    def __lt__(self, other):
        return False

    def __gt__(self, other):
        if other is None:
            return False

        return True


class Quantiles(Sequence):
    """
    A class representing quantiles (percentiles, quartiles, etc.) for a given
    column of Number data.
    """
    def __init__(self, quantiles):
        self._quantiles = quantiles

    def __getitem__(self, i):
        return self._quantiles.__getitem__(i)

    def __iter__(self):
        return self._quantiles.__iter__()

    def __len__(self):
        return self._quantiles.__len__()

    def __repr__(self):
        return repr(self._quantiles)

    def locate(self, value):
        """
        Identify which quantile a given value is part of.
        """
        i = 0

        if value < self._quantiles[0]:
            raise ValueError('Value is less than minimum quantile value.')

        if value > self._quantiles[-1]:
            raise ValueError('Value is greater than maximum quantile value.')

        if value == self._quantiles[-1]:
            return Decimal(len(self._quantiles) - 1)

        while value >= self._quantiles[i + 1]:
            i += 1

        return Decimal(i)


def median(data_sorted):
    """
    Finds the median value of a given series of values.

    :param data_sorted:
        The values to find the median of. Must be sorted.
    """
    length = len(data_sorted)

    if length % 2 == 1:
        return data_sorted[((length + 1) // 2) - 1]

    half = length // 2
    a = data_sorted[half - 1]
    b = data_sorted[half]

    return (a + b) / 2


def max_precision(values):
    """
    Given a series of values (such as a :class:`.Column`) returns the most
    significant decimal places present in any value.

    :param values:
        The values to analyze.
    """
    max_whole_places = 1
    max_decimal_places = 0
    precision = getcontext().prec

    for value in values:
        if value is None:
            continue

        sign, digits, exponent = value.normalize().as_tuple()

        exponent_places = exponent * -1
        whole_places = len(digits) - exponent_places

        if whole_places > max_whole_places:
            max_whole_places = whole_places

        if exponent_places > max_decimal_places:
            max_decimal_places = exponent_places

    # In Python 2 it was possible for the total digits to exceed the
    # available context precision. This ensures that can't happen. See #412
    if max_whole_places + max_decimal_places > precision:  # pragma: no cover
        max_decimal_places = precision - max_whole_places

    return max_decimal_places


def make_number_formatter(decimal_places, add_ellipsis=False):
    """
    Given a number of decimal places creates a formatting string that will
    display numbers with that precision.

    :param decimal_places:
        The number of decimal places
    :param add_ellipsis:
        Optionally add an ellipsis symbol at the end of a number
    """
    fraction = u'0' * decimal_places
    ellipsis = u'â€¦' if add_ellipsis else u''
    return u''.join([u'#,##0.', fraction, ellipsis, u';-#,##0.', fraction, ellipsis])


def round_limits(minimum, maximum):
    """
    Rounds a pair of minimum and maximum values to form reasonable "round"
    values suitable for use as axis minimum and maximum values.

    Values are rounded "out": up for maximum and down for minimum, and "off":
    to one higher than the first significant digit shared by both.

    See unit tests for examples.
    """
    min_bits = minimum.normalize().as_tuple()
    max_bits = maximum.normalize().as_tuple()

    max_digits = max(
        len(min_bits.digits) + min_bits.exponent,
        len(max_bits.digits) + max_bits.exponent
    )

    # Whole number rounding
    if max_digits > 0:
        multiplier = Decimal('10') ** (max_digits - 1)

        min_fraction = (minimum / multiplier).to_integral_value(rounding=ROUND_FLOOR)
        max_fraction = (maximum / multiplier).to_integral_value(rounding=ROUND_CEILING)

        return (
            min_fraction * multiplier,
            max_fraction * multiplier
        )

    max_exponent = max(min_bits.exponent, max_bits.exponent)

    # Fractional rounding
    q = Decimal('10') ** (max_exponent + 1)

    return (
        minimum.quantize(q, rounding=ROUND_FLOOR).normalize(),
        maximum.quantize(q, rounding=ROUND_CEILING).normalize()
    )


def letter_name(index):
    """
    Given a column index, assign a "letter" column name equivalent to
    Excel. For example, index ``4`` would return ``E``.
    Index ``30`` would return ``EE``.
    """
    letters = string.ascii_lowercase
    count = len(letters)

    return letters[index % count] * ((index // count) + 1)


def parse_object(obj, path=''):
    """
    Recursively parse JSON-like Python objects as a dictionary of paths/keys
    and values.

    Inspired by JSONPipe (https://github.com/dvxhouse/jsonpipe).
    """
    if isinstance(obj, dict):
        iterator = obj.items()
    elif isinstance(obj, (list, tuple)):
        iterator = enumerate(obj)
    else:
        return {path.strip('/'): obj}

    d = OrderedDict()

    for key, value in iterator:
        key = six.text_type(key)
        d.update(parse_object(value, path + key + '/'))

    return d


def issequence(obj):
    """
    Returns :code:`True` if the given object is an instance of
    :class:`.Sequence` that is not also a string.
    """
    return isinstance(obj, Sequence) and not isinstance(obj, six.string_types)


def slugify(values, ensure_unique=False, **kwargs):
    """
    Given a sequence of strings, returns a standardized version of the sequence.
    If ``ensure_unique`` is True, any duplicate strings will be appended with
    a unique identifier. Any kwargs will be passed to the Slugify or
    UniqueSlugify class constructor

    See: https://github.com/dimka665/awesome-slugify
    """
    # Default to all lowercase
    slug_args = {'to_lower': True}
    slug_args.update(kwargs)

    if ensure_unique:
        custom_slugify = UniqueSlugify(**slug_args)
    else:
        custom_slugify = Slugify(**slug_args)

    return tuple(custom_slugify(value) for value in values)
