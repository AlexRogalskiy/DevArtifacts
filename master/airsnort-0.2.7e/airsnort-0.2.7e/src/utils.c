/*
    This file is part of AirSnort.

    AirSnort is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    AirSnort is distributed in the hope that it will be useful,
      but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with AirSnort; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#ifndef WIN32
#include <semaphore.h>
#else
#include <windows.h>
#endif
#include "capture.h"

#ifndef WLAN_DEVNAMELEN_MAX
#ifndef WIN32
#define WLAN_DEVNAMELEN_MAX 16
#else
#define WLAN_DEVNAMELEN_MAX 128
#endif  //WIN32
#endif  //WLAN_DEVNAMELEN_MAX


static char rcname[] = "/.airsnortrc";

extern int breadth40;
extern int breadth128;
extern int cardType;
extern unsigned int chan;
extern unsigned int spinchan;
extern char *cards[];

#ifndef WIN32
sem_t malloc_sem; 
#else
HANDLE malloc_sem; 
#endif

void *malloc_r(unsigned int size) {
   void *result;
#ifndef WIN32
   sem_wait(&malloc_sem); 
#else
   WaitForSingleObject(malloc_sem, INFINITE); 
#endif
   result = malloc(size);
#ifndef WIN32
   sem_post(&malloc_sem);
#else
   ReleaseSemaphore(malloc_sem, 1, NULL); 
#endif
   return result;
}

void *calloc_r(unsigned int nmemb, unsigned int size) {
   void *result;
#ifndef WIN32
   sem_wait(&malloc_sem); 
#else
   WaitForSingleObject(malloc_sem, INFINITE); 
#endif
   result = calloc(nmemb, size);
#ifndef WIN32
   sem_post(&malloc_sem);
#else
   ReleaseSemaphore(malloc_sem, 1, NULL); 
#endif
   return result;
}

void free_r(void *ptr) {
#ifndef WIN32
   sem_wait(&malloc_sem); 
#else
   WaitForSingleObject(malloc_sem, INFINITE); 
#endif
   free(ptr);
#ifndef WIN32
   sem_post(&malloc_sem);
#else
   ReleaseSemaphore(malloc_sem, 1, NULL); 
#endif
}

//load user settings from $HOME/.airsnortrc
void loadOpts() {
   char *home = getenv("HOME");
   char *rcfile;
   char buf[128];
   FILE *rc;
   char *index;
   int r, i;
   unsigned int rclen;
   if (home) {
      rclen = strlen(home) + sizeof(rcname);
      rcfile = (char*) malloc_r(rclen);
      if (!rcfile) return;
#ifndef WIN32
      snprintf(rcfile, rclen, "%s%s", home, rcname);
#else
      _snprintf(rcfile, rclen, "%s%s", home, rcname);
#endif
      rc = fopen(rcfile, "r");
      if (rc) {
         while (fgets(buf, 128, rc)) {
            index = strchr(buf, '\n');
            if (index) *index = 0;
            index = strchr(buf, ':');
            if (index) {
               *index = 0;
               if (!strcmp(buf, "dev")) {
                  strncpy(dev, index + 1, WLAN_DEVNAMELEN_MAX);
                  dev[WLAN_DEVNAMELEN_MAX - 1] = 0;
               }
               else if (!strcmp(buf, "driver")) {
                  r = atoi(index + 1);
                  for (i = 0; cards[i]; i++);
                  cardType = (r < 0 || r >= i) ? 0 : r;
               }
               else if (!strcmp(buf, "breadth128")) {
                  r = atoi(index + 1);
                  breadth128 = (r < 1 || r > 20) ? 1 : r;
               }
               else if (!strcmp(buf, "breadth40")) {
                  r = atoi(index + 1);
                  breadth40 = (r < 1 || r > 20) ? 1 : r;
               }
               else if (!strcmp(buf, "channel")) {
                  r = atoi(index + 1);
                  chan = (r < 1 || r > 11) ? 6 : r;
                  spinchan = chan;
               }
            }
         }
         fclose(rc);
      }
      free_r(rcfile);
   }
}

//save user settings to $HOME/.airsnortrc
void saveOpts() {
   char *home = getenv("HOME");
   char *rcfile;
   FILE *rc;
   unsigned int rclen;
   if (home) {
      rclen = strlen(home) + sizeof(rcname);
      rcfile = (char*) malloc_r(rclen);
      if (!rcfile) return;
#ifndef WIN32
      snprintf(rcfile, rclen, "%s%s", home, rcname);
#else
      _snprintf(rcfile, rclen, "%s%s", home, rcname);
#endif
      rc = fopen(rcfile, "w");
      if (rc) {
         fprintf(rc, "dev:%s\n", dev);
         fprintf(rc, "driver:%d\n", cardType);
         fprintf(rc, "breadth128:%d\n", breadth128);
         fprintf(rc, "breadth40:%d\n", breadth40);
         fprintf(rc, "channel:%d\n", spinchan);
         fclose(rc);
      }
      free_r(rcfile);
   }
}
