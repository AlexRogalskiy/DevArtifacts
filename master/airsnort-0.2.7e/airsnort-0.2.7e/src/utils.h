/*
    This file is part of AirSnort.

    AirSnort is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    AirSnort is distributed in the hope that it will be useful,
      but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with AirSnort; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

#ifndef __AS_UTILS_H
#define __AS_UTILS_H

void *malloc_r(unsigned int size);
void *calloc_r(unsigned int nmemb, unsigned int size);
void free_r();

void loadOpts(void);
void saveOpts(void);

#endif

