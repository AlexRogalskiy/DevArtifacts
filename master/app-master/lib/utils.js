'use strict';

module.exports.removeFromArray = function(arr, item) {
	for (var i = arr.length - 1; i >= 0; i--) {
		if (arr[i] === item) {
			arr.splice(i, 1);
		}
	}
	return arr;
};


/**
 * Returns a function, that, when invoked, will only be triggered at most once 
 * during a given window of time. Normally, the throttled function will run 
 * as much as it can, without ever going more than once per wait duration; 
 * but if youâ€™d like to disable the execution on the leading edge, 
 * pass {leading: false}. To disable execution on the trailing edge, ditto.
 * 
 * @copyright Underscore.js
 */
module.exports.throttle = function(func, wait, options) {
	var context, args, result;
	var timeout = null;
	var previous = 0;
	if (!options) options = {};
	var later = function() {
		previous = options.leading === false ? 0 : Date.now();
		timeout = null;
		result = func.apply(context, args);
		if (!timeout) context = args = null;
	};
	return function() {
		var now = Date.now();
		if (!previous && options.leading === false) previous = now;
		var remaining = wait - (now - previous);
		context = this;
		args = arguments;
		if (remaining <= 0 || remaining > wait) {
			if (timeout) {
				clearTimeout(timeout);
				timeout = null;
			}
			previous = now;
			result = func.apply(context, args);
			if (!timeout) context = args = null;
		} else if (!timeout && options.trailing !== false) {
			timeout = setTimeout(later, remaining);
		}
		return result;
	};
};