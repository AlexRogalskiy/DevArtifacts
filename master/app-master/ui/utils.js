/**
 * Utils for appâ€™s UI part
 */
'use strict';

module.exports.qs = function(sel, context) {
	return (context || document).querySelector(sel);
};

module.exports.qsa = function(sel, context) {
	return toArray((context || document).querySelectorAll(sel));
};

module.exports.closest = function(elem, sel) {
	while (elem && elem !== document) {
		if (elem.matches(sel)) {
			return elem;
		}

		elem = elem.parentNode;
	}
};

function toArray(obj) {
	return Array.prototype.slice.call(obj);
}