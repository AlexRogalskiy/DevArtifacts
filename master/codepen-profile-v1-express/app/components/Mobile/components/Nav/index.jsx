/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import cx from 'classnames';
import { IndexLink, Link } from 'react-router';

/**
 * External dependencies
 */
import './style.scss';

/**
 * Local variables
 */
const propTypes = {
  className: PropTypes.string,
  items: PropTypes.array,
  menus: PropTypes.array,
};

const defaultProps = {};

/**
 * <Nav />
 */
class Nav extends Component {
  _renderItem(item, index) {
    let linkItem;

    if (item.index) {
      linkItem = (
        <IndexLink
          className="nav__link"
          activeClassName="nav__link--active"
          to={item.url}
        >
          <span className="nav__icon" dangerouslySetInnerHTML={{ __html: item.icon }} />
          <span dangerouslySetInnerHTML={{ __html: item.children }} />
        </IndexLink>
      );
    } else if (item.anchor) {
      linkItem = (
        <a
          className="nav__link"
          href={item.url}
        >
          <span className="nav__icon" dangerouslySetInnerHTML={{ __html: item.icon }} />
          <span dangerouslySetInnerHTML={{ __html: item.children }} />
        </a>
      );
    } else {
      linkItem = (
        <Link
          className="nav__link"
          activeClassName="nav__link--active"
          to={item.url}
        >
          <span className="nav__icon" dangerouslySetInnerHTML={{ __html: item.icon }} />
          <span dangerouslySetInnerHTML={{ __html: item.children }} />
        </Link>
      );
    }

    return (
      <li
        key={`mobile-nav-item-${index}`}
        className="nav__item"
      >
        {linkItem}
      </li>
    );
  }

  render() {
    let classes = cx('nav nav--mobile', this.props.className);

    return (
      <nav className={classes}>
        <ul className="nav__menu">
          {this.props.items.map(this._renderItem.bind(this))}
        </ul>
      </nav>
    );
  }
}

Nav.propTypes = propTypes;
Nav.defaultProps = defaultProps;

export default Nav;
