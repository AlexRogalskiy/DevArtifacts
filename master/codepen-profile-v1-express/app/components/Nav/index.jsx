/**
 * External dependencies
 */
import React, { Component, PropTypes } from 'react';
import cx from 'classnames';
import { IndexLink, Link } from 'react-router';

/**
 * External dependencies
 */
import './style.scss';

/**
 * Local variables
 */
const propTypes = {
  className: PropTypes.string,
  menus: PropTypes.array,
};

const defaultProps = {};

/**
 * <Nav />
 */
class Nav extends Component {
  _renderItem(item, index) {
    let linkItem;

    if (item.index) {
      linkItem = (
        <IndexLink
          className="nav__link"
          activeClassName="nav__link--active"
          to={item.url}
        >
          <span dangerouslySetInnerHTML={{ __html: item.children }} />
        </IndexLink>
      );
    } else if (item.anchor) {
      linkItem = (
        <a
          className="nav__link"
          href={item.url}
          dangerouslySetInnerHTML={{ __html: item.children }}
        />
      );
    } else {
      linkItem = (
        <Link
          className="nav__link"
          activeClassName="nav__link--active"
          to={item.url}
        >
          <span dangerouslySetInnerHTML={{ __html: item.children }} />
        </Link>
      );
    }

    return (
      <li
        key={`nav-item-${index}`}
        className="nav__item"
      >
        {linkItem}
      </li>
    );
  }

  _renderMenu(menu, index) {
    return (
      <ul
        key={`nav-menu-${index}`}
        className="nav__menu"
      >
        {menu.map(this._renderItem.bind(this))}
      </ul>
    );
  }

  render() {
    let classes = cx('nav', this.props.className);

    return (
      <nav className={classes}>
        {this.props.menus.map(this._renderMenu.bind(this))}
      </nav>
    );
  }
}

Nav.propTypes = propTypes;
Nav.defaultProps = defaultProps;

export default Nav;
