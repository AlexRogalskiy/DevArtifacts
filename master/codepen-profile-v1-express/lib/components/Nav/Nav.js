/**
 * External dependencies
 */
import React, { PropTypes } from 'react';
import { Link } from 'react-router';

/**
 * Internal dependencies
 */
import config from '../../../config';

/**
 * Local variables
 */
const propTypes = {
  type: PropTypes.number,
};

const defaultProps = {};

/**
 * <Nav />
 */
const Nav = ({ type }) => {
  let navItems;

  switch (type) {
    case 1:
      navItems = (
        <nav className="nav">
          <Link
            activeClassName="nav__item--active"
            className="nav__item"
            to="/shots"
          >
            Shots
          </Link>
          <Link
            activeClassName="nav__item--active"
            className="nav__item"
            to="/projects"
          >
            Projects
          </Link>
          {config.env === 'development' ? <Link activeClassName="nav__item--active" className="nav__item" to="/sandbox">Sandbox</Link> : undefined}
        </nav>
      );
      break;
    default:
      navItems = (
        <nav className="nav">
          <Link
            activeClassName="nav__item--active"
            className="nav__item"
            to="/pens"
          >
            Pens
          </Link>
          <Link
            activeClassName="nav__item--active"
            className="nav__item"
            to="/posts"
          >
            Posts
          </Link>
          <Link
            activeClassName="nav__item--active"
            className="nav__item"
            to="/collections"
          >
            Collections
          </Link>
          {config.env === 'development' ? <Link activeClassName="nav__item--active" className="nav__item" to="/sandbox">Sandbox</Link> : undefined}
        </nav>
      );
      break;
  }

  return navItems;
};

Nav.propTypes = propTypes;
Nav.defaultProps = defaultProps;

export default Nav;
