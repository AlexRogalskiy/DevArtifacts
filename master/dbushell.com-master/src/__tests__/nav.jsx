import React from 'react';
import Nav from '../components/nav';
import {outerHTML} from './utils';

describe('Nav component', () => {
  it(`renders with defaults`, () => {
    expect(
      outerHTML(
        <Nav/>
      )
    ).toMatchSnapshot();
  });
});
