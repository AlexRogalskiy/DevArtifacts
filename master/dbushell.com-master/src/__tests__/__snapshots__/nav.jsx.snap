// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders with defaults 1`] = `
"<nav class=\\"b-nav\\" id=\\"nav\\">
  <h2 class=\\"b-nav__title u-vh\\">Website Navigation</h2>
  <ul class=\\"b-nav__list\\">
    <li data-id=\\"home\\" class=\\"b-nav__item\\"><a href=\\"/\\" class=\\"b-nav__link\\">Home</a></li>
    <li data-id=\\"about\\" class=\\"b-nav__item\\"><a href=\\"/about/\\" class=\\"b-nav__link\\">About</a></li>
    <li data-id=\\"services\\" class=\\"b-nav__item\\"><a href=\\"/services/\\" class=\\"b-nav__link\\">Services</a></li>
    <li data-id=\\"portfolio\\" class=\\"b-nav__item\\"><a href=\\"/showcase/\\" class=\\"b-nav__link\\">Portfolio</a></li>
    <li data-id=\\"blog\\" class=\\"b-nav__item\\"><a href=\\"/blog/\\" class=\\"b-nav__link\\">Blog</a></li>
    <li data-id=\\"contact\\" class=\\"b-nav__item\\"><a href=\\"/contact/\\" class=\\"b-nav__link\\">Contact</a></li>
    <li data-id=\\"icons\\" class=\\"b-nav__item b-nav__item--icons\\"><a class=\\"b-nav__link\\" rel=\\"me noopener noreferrer\\" title=\\"David Bushell on Twitter\\" href=\\"https://twitter.com/dbushell\\" target=\\"_blank\\"><svg><use xlink:href=\\"/assets/img/icons.svg#twitter\\"></use></svg><span class=\\"u-vh\\">@dbushell</span></a><a class=\\"b-nav__link\\"
        rel=\\"me noopener noreferrer\\" title=\\"David Bushell on GitHub\\" href=\\"https://github.com/dbushell/\\" target=\\"_blank\\"><svg><use xlink:href=\\"/assets/img/icons.svg#github\\"></use></svg><span class=\\"u-vh\\">GitHub</span></a><a class=\\"b-nav__link\\" rel=\\"me noopener noreferrer\\"
        title=\\"David Bushell on CodePen\\" href=\\"https://codepen.io/dbushell/\\" target=\\"_blank\\"><svg><use xlink:href=\\"/assets/img/icons.svg#codepen\\"></use></svg><span class=\\"u-vh\\">CodePen</span></a></li>
  </ul>
</nav>"
`;
